### PR TITLE
HRINT-2675 - FTSQuerySearchOperators.java

### PR DESCRIPTION
**Pull request by Drorlev**

### CODE REVIEW

**Review by ml054 — APPROVED** [review]
java part looks good

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/metadata.json [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        {
+            "Title": "Define search criteria I",
+            "Slug": "step-2",
+            "DescriptionLines": [
+                "* Use `search()` to search for the specified terms on the specified document field.",
+                "",
+                "* The default operator between 'term1' and 'term2' is OR.<br>",
+                "  Use `SearchOperator.AND` to set the logical AND operation instead.",
+
+                "* In this Search, we search for 'term1' and 'term2' in the document field _Notes_."
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/metadata.json:61]: `'term1' and 'term2'` => `_term1 AND term2_`

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/metadata.json [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        },
+        {
+            "Title": "Define search criteria II",
+            "Slug": "step-3",
+            "DescriptionLines": [
+                "* Another `search()` method can be called for searching on another document field or terms.",
+                "",
+                "* The specified logical operation is applied between this 'Search' and the previous 'Search'.",
+                "",
+                "* With this Search added, we are now searching for ('term1' and 'term2' ) OR _term3_ in the document field _Notes_."
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/metadata.json:86]: `( 'term1' and 'term2' )` => `( _term1_ AND _term2_ )`

**Review by Danielle9897 — CHANGES_REQUESTED** [review]

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/demos.json [hunk 3]
```diff
@@ -100,6 +100,7 @@
         },
         "demos": [
             "fTSWithStaticIndexSingleField",
+            "fTSQuerySearchOperators",
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/demos.json:103]: place before the static indexes demos - keep order as in C#

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/FTSQuerySearchOperators.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        String term2 = runParams.getTerm2();
+        String term3 = runParams.getTerm3();
+
+        //region Demo
+        List<Employee> employeesWithMatchingTerms;
+
+        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
+            //region Step_1
+            employeesWithMatchingTerms = session.query(Employee.class)
+                //endregion
```
[the remark this answers is not in the document]
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/FTSQuerySearchOperators.java:23]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/FTSQuerySearchOperators.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                //region Step_2
+                .search("Notes", term1 + " " + term2, SearchOperator.AND)
+                //endregion
+                //region Step_3
+                .search("Notes", term3, SearchOperator.OR)
+                //endregion
+                //region Step_4
+                .toList();
+            //endregion
```
[the remark this answers is not in the document]
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/FTSQuerySearchOperators.java:32]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/FTSQuerySearchOperators.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
+package net.ravendb.demo.textSearch.FTSQuerySearchOperators;
+
+import net.ravendb.client.documents.queries.SearchOperator;
+import net.ravendb.client.documents.session.IDocumentSession;
+import net.ravendb.demo.common.DocumentStoreHolder;
+import net.ravendb.demo.common.models.Employee;
+import net.ravendb.demo.textSearch.fTSQuerySearchBasics.FTSQuerySearchBasics;
+
+import java.util.List;
```
[the remark this answers is not in the document]
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/FTSQuerySearchOperators.java:9]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/metadata.json (hunk 2 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/metadata.json:86]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/metadata.json (hunk 1 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/metadata.json:61]: Done

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/demos.json (hunk 3 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/demos.json:104]: Done

**Review by Danielle9897 — APPROVED** [review]

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/demos.json (hunk 3 above)
Comment (Danielle9897) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/demos.json:104]: Pls keep the same order as in the C# demos.json file
The file is located in `Controlers\Demos`

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/metadata.json [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        "**You can customize the logical operators** between the searched terms.",
+        "",
+        "Upon query, an auto-index is created on the searched document fields."
+    ],
+
+    "Assets": [
+        {
+            "Title": "Operation in queries - Book",
+            "Type": "link",
+            "Url": "https://ravendb.net/learn/inside-ravendb-book/reader/4.0/9-querying-in-ravendb#operations-in-queries"
```
Comment (Danielle9897) [inline DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/metadata.json:17]: As discussed when you first started to work on the demos 
Pls align the `':'`

**Review by Danielle9897 — CHANGES_REQUESTED** [review]

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/metadata.json (hunk 7 above)
Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/metadata.json]: Done

Comment (Drorlev) [reply DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSQuerySearchOperators/metadata.json]: "Pls keep the same order as in the C# demos.json file
The file is located in Controlers\Demos" - also Done

**Review by Danielle9897 — APPROVED** [review]

**Comment by Danielle9897** [comment]
@marcin this can be merged

**Pull request merged**